packages/application-shell/src/components/application-shell/application-shell.js

```diff
@@ -222,11 +222,12 @@ export const RestrictedApplication = props => (
                             <FetchProject projectKey={projectKeyFromUrl}>
                               {({ isLoading: isLoadingProject, project }) => {
                                 // Render the loading navbar as long as all the data
-                                // hasn't been loaded.
+                                // hasn't been loaded, or if the project does not exist.
                                 if (
                                   isLoadingUser ||
                                   isLoadingLocaleData ||
-                                  isLoadingProject
+                                  isLoadingProject ||
+                                  !project
```

Comment:
This is to avoid rendering the navbar. If we don't do that, the query to the settings api to fetch the custom app links will fail with `401`, because the `projectKey` is missing, causing the app to log the user out.
So we only render the navbar if there is a project

Reply:
First of all sorry for point something that was not added in this PR but would make sense to extract this condition into a variable for the sake of readability `canRenderNavBar` or something like that? 😋